media/js/mozorg/home/takeover.js

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+$(function () {
+    'use strict';
+
+    var $takeoverModal = $('#fundraising_takeover');
+    var $overlay = $('.page-overlay');
+    var $closeModal = $('#close_takeover');
+    var $form = $('#fundraiser-form');
+
+    var $donateOptionsSection = $('.donate-options');
+    var $donateCustom = $('#donate-custom');
+    var $optionLabels = $('label[class!="own-amount"]', $donateOptionsSection);
+    var $options = $('input[type="radio"]', $donateOptionsSection);
+
+    // sets the mozorg.takeover state to true indicating that the user has
+    // seen the takeover modal during their current session.
+    function storeSession() {
+        try {
+            sessionStorage.setItem('mozorg.takeover.seen', 'true');
+        } catch (ex) {
+            // yum, errors taste nice ;)
+        }
+    }
+
+    // Show and hides the takeover modal
+    // Removes the keyup event from body if specified
+    function toggleModal(removeListener) {
+
+        if(removeListener) {
+            // no longer listen for keyup events.
+            $('body').off('keyup.takeover');
+        }
+
+        $([$takeoverModal, $overlay]).each(function() {
+            $(this).toggleClass('hidden');
+        });
+    }
+
+    // removes checked from all the radio buttons
+    // and selected from the labels.
+    function clearOptions() {
+        $optionLabels.removeClass('selected');
+        $options.removeAttr('checked');
+    }
+
+    // shows the take over modal, sets the required entry in
+    // session storage and adds an `esc` key handler to close
+    // the modal.
+    function initTakeOver() {
+        // show the takeover
+        toggleModal();
+        // set mozorg.takeover.seen to true
+        storeSession();
+
+        // trap key events, listening for escape.
+        // we only want to do this here so, we do not attach the event
+        // if we are not showing the modal.
+        $('body').on('keyup.takeover', function(event) {
```

Comment:
We could also cache `$('body')` as a variable since we're also removing the event later on?